File: frontend-web/src/services/notificationsService.js
/**
 * Service pour gérer les notifications
 */
import apiClient from './api';

export const notificationsService = {
  /**
   * Lister les notifications
   */
  async listNotifications(page = 1, limit = 20, unreadOnly = false) {
    const response = await apiClient.get('/notifications', {
      params: { page, limit, unread_only: unreadOnly },
    });
    return response.data;
  },

  /**
   * Obtenir le nombre de notifications non lues
   */
  async getUnreadCount() {
    const response = await apiClient.get('/notifications/unread-count');
    return response.data;
  },

  /**
   * Marquer une notification comme lue
   */
  async markAsRead(notificationId) {
    const response = await apiClient.patch(`/notifications/${notificationId}/read`);
    return response.data;
  },

  /**
   * Marquer toutes les notifications comme lues
   */
  async markAllAsRead() {
    const response = await apiClient.post('/notifications/mark-all-read');
    return response.data;
  },

  /**
   * Supprimer une notification
   */
  async deleteNotification(notificationId) {
    const response = await apiClient.delete(`/notifications/${notificationId}`);
    return response.data;
  },
};

File: mobile-app/lib/services/notifications_service.dart
/// Service pour gérer les notifications
import '../services/api_service.dart';
import 'package:dio/dio.dart';

class NotificationsService {
  final ApiService _apiService = ApiService();

  /// Lister les notifications
  Future<Response> listNotifications({
    int page = 1,
    int limit = 50,
    bool? unreadOnly,
  }) async {
    final queryParams = {
      'page': page.toString(),
      'limit': limit.toString(),
      if (unreadOnly != null) 'unread_only': unreadOnly.toString(),
    };
    return await _apiService.get('/notifications', queryParameters: queryParams);
  }

  /// Marquer une notification comme lue
  Future<Response> markAsRead(String notificationId) async {
    return await _apiService.patch('/notifications/$notificationId/read');
  }

  /// Marquer toutes les notifications comme lues
  Future<Response> markAllAsRead() async {
    return await _apiService.post('/notifications/read-all');
  }

  /// Supprimer une notification
  Future<Response> deleteNotification(String notificationId) async {
    return await _apiService.delete('/notifications/$notificationId');
  }

  /// Obtenir le nombre de notifications non lues
  Future<Response> getUnreadCount() async {
    return await _apiService.get('/notifications/unread-count');
  }
}


